### PR TITLE
Decompile unk_02090C94

### PR DESCRIPTION
**Pull request by abaresk**

### CODE REVIEW

File: src/mail_misc.c [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        return;
+    case 1:
+        msg1->msg_bank = a0->unk1C[0];
+        msg1->msg_no = a0->unk1C[1];
+        return;
+    case 2:
+        msg2->msg_bank = a0->mailMessage.msg_bank;
+        msg2->msg_no = a0->mailMessage.msg_no;
+        msg2->fields[0] = a0->mailMessage.fields[0];
+        msg2->fields[1] = a0->mailMessage.fields[1];
```
Comment (tgsm) [inline src/mail_misc.c:125]:
```suggestion
        *msg2 = a0->mailMessage;
```

File: src/mail_misc.c [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+    u16 *src = (u16 *)&msg1->msg_bank;
+    for (int i = 0; i < 2; i++) {
+        a0->unk1C[i] = src[i];
+    }
+
+    a0->mailMessage.msg_bank = msg2->msg_bank;
+    a0->mailMessage.msg_no = msg2->msg_no;
+    a0->mailMessage.fields[0] = msg2->fields[0];
+    a0->mailMessage.fields[1] = msg2->fields[1];
```
Comment (tgsm) [inline src/mail_misc.c:154]:
```suggestion
    a0->mailMessage = *msg2;
```

File: src/mail_misc.c [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+}
+
+int sub_02090E44(Unk02090C94 *a0) {
+    if (a0 == NULL || a0->unk24 == 0) return 0;
```
Comment (tgsm) [inline src/mail_misc.c:158]:
```suggestion
    if (a0 == NULL || a0->unk24 == 0) {
        return 0;
    }
```

File: src/unk_02091054.c [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+u8 sub_02091084(u32 idx) {
+    GF_ASSERT(idx < NELEMS(_02105F04));
+    return _02105F04[idx].unk2;
+}
+
+BOOL sub_0209109C(u32 idx) {
+    GF_ASSERT(idx < NELEMS(_02105F04));
+    if (idx < 0x32 || idx > 0x4d) return FALSE;
```
Comment (tgsm) [inline src/unk_02091054.c:30]:
```suggestion
    if (idx < 0x32 || idx > 0x4d) {
        return FALSE;
    }
```

File: src/unk_02091054.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    return TRUE;
+}
+
+u32 sub_020910B8(u32 idx) {
+    GF_ASSERT(idx < NELEMS(_02105F04));
+    Unk02091054 *ptr = &_02105F04[idx];
+    return ((u32)(ptr->unk3 << 0x18) >> 0x1a);
```
Comment (tgsm) [inline src/unk_02091054.c:38]: This smells like a bitfield

File: src/mail_misc.c [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+void sub_02090D18(Unk02090C94 *a0, u16 a1, u16 a2) {
+    a0->unk1C[0] = a1;
+    a0->unk1C[1] = a2;
+}
+
+void sub_02090D20(Unk02090C94 *a0, MAIL_MESSAGE *mailMessage) {
+    a0->mailMessage.msg_bank = mailMessage->msg_bank;
+    a0->mailMessage.msg_no = mailMessage->msg_no;
+    a0->mailMessage.fields[0] = mailMessage->fields[0];
+    a0->mailMessage.fields[1] = mailMessage->fields[1];
```
Comment (tgsm) [inline src/mail_misc.c:51]:
```suggestion
    a0->mailMessage = *mailMessage;
```

**Review by tgsm — CHANGES_REQUESTED** [review]

File: src/mail_misc.c [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+    ptr->mailbox = mailbox;
+    ptr->unk0[0] = 1;
+    ptr->unk8 = a1;
+    ptr->unkC = 0;
+    ptr->saveData = saveData;
+
+    MAIL *mail = Mail_new(heapId);
+    ptr->mail = mail;
+    Mail_init(mail);
+    Mail_SetNewMessageDetails(ptr->mail, -1, partyIdx, saveData);
```
Comment (PikalaxALT) [inline src/mail_misc.c:176]:
```suggestion
    Mail_SetNewMessageDetails(ptr->mail, MAIL_NONE, partyIdx, saveData);
```

File: src/unk_02091054.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+typedef struct {
+    u8 unk0;
+    u8 unk1;
+    u8 unk2;
+    u8 unk3_0:2;
+    u8 unk3_2:6;
+} Unk02091054;
+
+extern Unk02091054 _02105F04[81];
```
Comment (PikalaxALT) [inline src/unk_02091054.c:11]:
```suggestion
extern const Unk02091054 _02105F04[81];
```

**Review by PikalaxALT — DISMISSED** [review]

File: src/unk_02091054.c (hunk 8 above)
Comment (PikalaxALT) [reply src/unk_02091054.c]: Now that you've marked this `const`, you can port the data from asm to here.

**Pull request merged**